tests/integration/test.changes.js

```diff
@@ -746,6 +746,86 @@ adapters.forEach(function (adapter) {
       });
     });
 
+    it('Immediately cancel changes', function () {
+      // fixes code coverage by ensuring the changes() listener
+      // emits 'complete' even if the db's task queue isn't
+      // ready yet
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        var changes = db.changes({live: true});
+        changes.on('error', reject);
+        changes.on('complete', resolve);
+        changes.cancel();
+      });
+    });
+
+    it('Changes with invalid ddoc view name', function () {
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        db.post({});
+        var changes = db.changes({live: true, filter: '_view', view: ''});
+        changes.on('error', resolve);
+        changes.on('change', reject);
+      });
+    });
+
+    it('Changes with invalid ddoc view name 2', function () {
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        db.post({});
+        var changes = db.changes({live: true, filter: '_view', view: 'a/b/c'});
+        changes.on('error', resolve);
+        changes.on('change', reject);
+      });
+    });
+
+    if (adapter === 'local') {
+      // This test crashes CouchDB due to
+      // https://issues.apache.org/jira/browse/COUCHDB-2765
+      it('Changes with invalid ddoc with no map function', function () {
+        var db = new PouchDB(dbs.name);
+        return db.put({
+          _id: '_design/name',
+          views: {
+            name: {
+              empty: 'sad face'
+            }
+          }
+        }).then(function () {
+          return new PouchDB.utils.Promise(function (resolve, reject) {
+            var changes = db.changes({
+              live: true,
+              filter: '_view',
+              view: 'name/name'
+            });
+            changes.on('error', resolve);
+            changes.on('change', reject);
+          });
+        });
+      });
+    }
```

Comment:
@daleharvey  You're talking about this thing? Yeah, the only reason I added it was because it improves the code coverage, since there's a line in `changes.js` where we say `var mapFun = ddoc && ddoc.views && ddoc.views[viewName[1]] && ddoc.views[viewName[1]].map`. I don't think there's any way to improve coverage without making a Pouch-only test, since CouchDB breaks for the same test.